crates/cast/src/script.rs

```diff
@@ -0,0 +1,365 @@
+        clippy::trivially_copy_pass_by_ref,
+        clippy::too_many_arguments
+    )]
+    fn match_cheatcode_by_selector(
+        &mut self,
+        vm: &mut VirtualMachine,
+        selector: &str,
+        inputs: &[Felt252],
+        output_start: &CellRef,
+        output_end: &CellRef,
+    ) -> Result<(), EnhancedHintError> {
+        let mut buffer = MemBuffer::new_segment(vm);
+        let result_start = buffer.ptr;
+
+        match selector {
+            "call" => {
+                let contract_address = inputs[0].to_field_element();
+                let function_name = as_cairo_short_string(&inputs[1])
+                    .expect("Failed to convert function name to short string");
+                let calldata_length = inputs[2]
+                    .to_usize()
+                    .expect("Failed to convert calldata length to usize");
+                let calldata = Vec::from(&inputs[3..(3 + calldata_length)]);
+                let calldata_felts: Vec<FieldElement> = calldata
+                    .iter()
+                    .map(StarknetConversions::to_field_element)
+                    .collect();
+
+                let call_response = self.runtime.block_on(call::call(
+                    contract_address,
+                    &function_name,
+                    calldata_felts,
+                    self.provider,
+                    &BlockId::Tag(Pending),
+                ))?;
+
+                buffer
+                    .write(call_response.data.len())
+                    .expect("Failed to insert data length");
+
+                buffer
+                    .write_data(
+                        call_response
+                            .data
+                            .iter()
+                            .map(StarknetConversions::to_felt252),
+                    )
+                    .expect("Failed to insert data");
+
+                Ok(())
+            }
+            _ => Err(anyhow!("Unknown cheatcode selector: {selector}")),
+        }?;
+
+        let result_end = buffer.ptr;
+        insert_value_to_cellref!(vm, output_start, result_start)?;
+        insert_value_to_cellref!(vm, output_end, result_end)?;
+
+        Ok(())
+    }
+}
+
+pub fn run(
+    script_path: &Utf8PathBuf,
+    provider: &JsonRpcClient<HttpTransport>,
+    runtime: Runtime,
+    scarb_manifest_path: &Utf8PathBuf,
+) -> Result<ScriptResponse> {
+    ScarbCommand::new().arg("build").run()?;
+
+    let metadata = get_scarb_metadata(scarb_manifest_path, true)?;
+    let package_metadata = get_package_metadata(&metadata, scarb_manifest_path)?;
+
+    let filename = format!("{}.sierra.json", package_metadata.name);
+    let path = metadata
+        .target_dir
+        .unwrap_or(metadata.workspace.root.join("target"))
+        .join(metadata.current_profile)
+        .join(filename.clone());
+
+    ensure!(
+        path.exists(),
+        formatdoc! {r#"
+            package has not been compiled, file does not exist: {filename}
```

Comment:
Consider showing full path instead of filename only.

Reply:
done